src/vs/workbench/common/editor/untitledEditorModel.ts

```diff
@@ -115,7 +115,9 @@ export class UntitledEditorModel extends BaseTextEditorModel implements IEncodin
 		return this.dirty;
 	}
 
-	private setDirty(dirty: boolean): void {
+	// {{SQL CARBON EDIT}}
```

Comment:
I think you can do this without needing to change vscode code by setting the initial value in the input rather than setting the value in the model after the editor is creating.